libs/librepcb/projecteditor/boardeditor/fsm/bes_select.cpp

```diff
@@ -309,259 +338,137 @@ BES_Base::ProcRetVal BES_Select::processIdleSceneRightMouseButtonReleased(
         QAction* a = aChangeFootprintMenu->addAction(
             footprintIcon, *footprint.getNames().value(
                                mProject.getSettings().getLocaleOrder()));
-        a->setData(footprint.getUuid().toStr());
         if (footprint.getUuid() ==
             devInst.getFootprint().getLibFootprint().getUuid()) {
           a->setCheckable(true);
           a->setChecked(true);
           a->setEnabled(false);
+        } else {
+          connect(a, &QAction::triggered, [this, board, &devInst, footprint](){
```

Comment:
I think `footprint` can be passed by reference as well, to avoid copying each footprint object by value:

```suggestion
          connect(a, &QAction::triggered, [this, board, &devInst, &footprint](){
```